app/models/pipeline_run.rb

```diff
@@ -1038,6 +1057,32 @@ def compile_stats_file
     save
   end
 
+  # Fetch the unmapped reads count from alignment stage then refined counts from
+  # assembly stage, as each becomes available. Prior to Dec 2019, the count was
+  # only fetched from alignment.
+  def fetch_unmapped_reads(
+    all_counts,
+    s3_path = "#{postprocess_output_s3_path}/#{ASSEMBLY_PREFIX}#{DAG_ANNOTATED_COUNT_BASENAME}"
+  )
+    unmapped_reads = nil
+    unidentified = all_counts.detect { |entry| entry.value?("unidentified_fasta") }
+    if unidentified
+      unmapped_reads = unidentified[:reads_after]
+
+      if supports_assembly? && finalized?
```

Comment:
curious, why is this clause inside the `if unidentified` block?

Reply:
I guess the idea is that if the alignment unmapped read count isn't available, then there's no way the assembly umapped read count is available, so we can skip it. Is it possible for the alignment one to be missing, but the assembly one to be present?

Reply:
Yes just like mark said.

Reply:
Yes just like mark said.